test/test-time-profiler.ts

```diff
@@ -0,0 +1,89 @@
+import * as delay from 'delay';
+import {perftools} from '../src/profile';
+import {TimeProfiler} from '../src/profilers/time-profiler';
+import {getIndexOrAdd} from '../src/util';
+
+let assert = require('assert');
+
+describe('TimeProfiler', function() {
+  describe('profile', function() {
+    it('should profile during duration and finish profiling after duration',
+       function() {
+         const duration = 500;
+         const sampleInterval = 1000;
+         let profiler = new TimeProfiler(sampleInterval);
+         let profilePromise = profiler.profile(duration);
+         return delay(2 * duration).then(() => {
```

Comment:
Making the test function async and then use `await` instead of `then`.

Here and elsewhere.